stimela/tests/test_cubical_recipe.yml

```diff
@@ -13,29 +13,45 @@ cabs:
         required: false
         policies:
           positional_head: true
+      dummy:
+        choices: [a, b]
 
   wsclean:
     image: null
     command: echo # just a dummy for now 
 
 recipe:
-  name: "cubical-image"
-  info: 'does one step of cubical, followed by one step of imaging'
-  dirs:
-    log: logs
-  aliases:
-    ms: [calibrate.ms, image.ms]
-    image: image.prefix
-  defaults:
-    calibrate_parset: solve-x.parset
-    image: foo
-    image_size: 10000
-    image_scale: 0.8asec
+  name: "cubical-loop"
+  for_loop:
+    var: ms
+    over: ms_list
+  inputs:
+    ms_list:
+      dtype: List[MS]
+      required: true
+  steps:
+    cubical-image:
+      recipe:
+        name: "cubical-image"
+        info: 'does one step of cubical, followed by one step of imaging'
+        dirs:
+          log: logs
+        aliases:
+          ms: [calibrate.ms, image.ms]
```

Comment:
As I noted in the discussion earlier today, you have dropped the "{ }" from values that are references. Is this intentional?

Reply:
No, it's just that during implementation I realized that there are two kinds of references: substitutions, and aliases.

Substitutions are just "dumb" string {}-substitutions. Something like "{calibrate.foo}_{image.bar}" just dumbly evaluates to a string (and may then be checked against some schema, depending on context). 

Aliases are stronger, formal "is a" relationship. In the example above, the ``ms`` parameter of the recipe will automatically get the same schema as ``calibrate.ms`` and ``image.ms``. 

If you look at ``test_recipe.yml``, it shows both kinds of references.